packages/firebase_auth/lib/firebase_auth.dart

```diff
@@ -165,15 +165,19 @@ typedef void PhoneCodeSent(String verificationId, [int forceResendingToken]);
 typedef void PhoneCodeAutoRetrievalTimeout(String verificationId);
 
 class FirebaseAuth {
-  FirebaseAuth({FirebaseApp app})
-      : app = app != null ? app : FirebaseApp.instance {
+  FirebaseAuth._(this.app) {
     channel.setMethodCallHandler(_callHandler);
   }
 
-  FirebaseAuth._() {
-    channel.setMethodCallHandler(_callHandler);
+  /// Provides an instance of this class corresponding to `app`.
+  factory FirebaseAuth.fromApp(FirebaseApp app) {
```

Comment:
After discussing with @kroikie, this is the way we will add multi-app support for our Firebase plugins.